src/base/bittorrent/session.cpp

```diff
@@ -775,12 +775,12 @@ bool Session::editCategory(const QString &name, const QString &savePath)
     m_categories[name] = savePath;
     m_storedCategories = map_cast(m_categories);
     if (isDisableAutoTMMWhenCategorySavePathChanged()) {
-        foreach (TorrentHandle *const torrent, torrents())
+        for (TorrentHandle *const torrent : torrents())
```

Comment:
Why didn't you apply `asConst()` to all `torrents()` calls?

Reply:
Clazy doesn't complain about possible detachment here.

Reply:
>Clazy doesn't complain about possible detachment on them.

But it REALLY does! And it is REAL detachment in this case (unlike of `mid()` which Clazy complains about).
Clazy gives too many false positives and incorrect results for us to blindly believe it.

Reply:
Oh no... there are other places i haven't added asConst. Do i just add it everywhere then?

Reply:
>whether i should add asConst to the ones that don't need it.

If you can identify them for sure...
Adding asConst to the ones that don't need it is less evil than skipping the ones that are prone to the detaching problem.

Reply:
I can, as you said if the container is created in the function then it's not shared. But if the function changes, the for() loops will surely be forgotten.

Reply:
>Adding asConst to the ones that don't need it is less evil than skipping the ones that are prone to the detaching problem.

Agree.